qa-tests-backend/src/test/groovy/Services.groovy

```diff
@@ -222,6 +222,22 @@ class Services extends BaseService {
         return violations == null || violations.size() == 0
     }
 
+    static boolean expectNoViolations(String deploymentName, String policyName, int timeoutSeconds = 60) {
+        int intervalSeconds = 3
+        int retries = (timeoutSeconds / intervalSeconds).intValue()
+        String query = "Deployment:${deploymentName}+Policy:${policyName}"
+
+        Timer t = new Timer(retries, intervalSeconds)
+        while (t.IsValid()) {
+            def violations = AlertService.getViolations(ListAlertsRequest.newBuilder().setQuery(query).build())
```

Comment:
If we get an exception here we will not retry

Reply:
I'm going to say that is OK because in the test `waitForViolation()` is used which ultimately makes the same call without retrying on exception and we do not see flakes due to that exception there.